models/codec_config_builder.go

```diff
@@ -0,0 +1,71 @@
+package models
+
+import "github.com/bitmovin/bitmovin-go/bitmovintypes"
+
+type H264CodecConfigBuilder struct {
+	Config *H264CodecConfiguration
+}
+
+func NewH264CodecConfigBuilder(name string) *H264CodecConfigBuilder {
+	return &H264CodecConfigBuilder{
+		Config: &H264CodecConfiguration{
+			Name: &name,
+		},
+	}
+}
+
+func (b *H264CodecConfigBuilder) Width(w int64) *H264CodecConfigBuilder {
+	b.Config.Width = &w
+	return b
+}
```

Comment:
I like this a lot.  It allows for compact setup of any sort of struct and helps with fields that necessarily need to be nullable ints or strings without having to convert them explicitly.